File: app/src/main/java/io/github/sithengineer/motoqueiro/ui/home/HomeContract.java
package io.github.sithengineer.motoqueiro.ui.home;

import rx.Observable;

public interface HomeContract {

  interface View extends io.github.sithengineer.motoqueiro.View {

    Observable<Void> handleStartRideClick();

    void sendToGpsSettings();

    void showActivateGpsViewMessage();

    void showGenericError();

    void showMiBandAddress(String miBandAddress);

    Observable<String> getMiBandAddressChanges();

    void showMiBandAddressError();

    void cleanMiBandAddressError();

    void showGivePermissionsMessage();

    String getRideName();

    int getSelectedDevicePosition();
  }

  interface Presenter extends io.github.sithengineer.motoqueiro.Presenter {
    void showActivateGpsViewMessage();

    void showActivateGpsView();
  }
}

File: app/src/main/java/io/github/sithengineer/motoqueiro/ui/home/HomeContract.kt
package io.github.sithengineer.motoqueiro.ui.home

import io.reactivex.Observable

interface HomeContract {

  interface View : io.github.sithengineer.motoqueiro.ui.View {

    fun miBandAddressChanges(): Observable<String>

    fun selectedDevicePosition(): Int

    fun getRideName(): String

    fun handleStartRideClick(): Observable<Any>

    fun sendToGpsSettings()

    fun showActivateGpsViewMessage()

    fun showGenericError()

    fun showMiBandAddress(miBandAddress: String)

    fun showMiBandAddressError()

    fun cleanMiBandAddressError()

    fun showGivePermissionsMessage()
  }

  interface Presenter : io.github.sithengineer.motoqueiro.ui.Presenter {
    fun showActivateGpsViewMessage()

    fun showActivateGpsView()
  }
}
